.github/workflows/ruskwallet_build.yml

```diff
@@ -0,0 +1,86 @@
+name: Compile CLI wallet binaries
+
+on:
+  workflow_dispatch:
+    inputs:
+      dusk_blockchain_ref:
+        description: "GIT branch, ref, or SHA to checkout"
+        required: true
+        default: "main"
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build_and_publish:
+    name: Build rusk-wallet binaries for ${{ matrix.os }} with ${{ matrix.compiler }}.
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-20.04, ubuntu-22.04, macos-latest, macos-11, windows-latest]
+        compiler: [cargo]
+        include:
+          - os: ubuntu-20.04
+            compiler: cargo
+            target: linux-x64
+
+          - os: ubuntu-22.04
+            compiler: cargo
+            target: linux-x64-libssl3
+
+          - os: macos-latest
+            compiler: cargo
+            target: macos-intel
+
+          - os: macos-11
+            compiler: cargo
+            target: macos-arm64
+            flags: --target=aarch64-apple-darwin
+            platform: aarch64-apple-darwin
+
+          - os: windows-latest
+            compiler: cargo
+            target: windows-x64
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.inputs.dusk_blockchain_ref }}
+
+      - name: Install dependencies
+        uses: dsherret/rust-toolchain-file@v1
+
+      - name: Add arm target for Apple Silicon build
+        run: rustup target add aarch64-apple-darwin
+        if: ${{ matrix.os == 'macos-11' }}
+
+      - name: Build Wallet
+        shell: bash
+        working-directory: ./
+        run: ${{matrix.compiler}} b --release --verbose ${{matrix.flags}}
```

Comment:
Can we just build the inner rusk-wallet module instead of the whole repository?

Reply:
> Can we just build the inner rusk-wallet module instead of the whole repository?

I was told this would work but I highly doubt it, unless we use a make command for the CLI or we set it to `./rusk-wallet`. In the latter case, the `target/release` folder should still work though. Even if you're in the subdir, it should still place the target in the root folder. And the following steps reset the working directory.

Reply:
Lets merge this PR and run the dispatch to see where we are